source/contribute.rst

```diff
@@ -41,128 +41,150 @@ contributing back to a Doctrine project.
 Initial Setup
 -------------
 
+-  Clone the project
 -  Setup a `GitHub <https://github.com>`_ account.
 -  Fork the repository of the project you want to contribute to. In this example
    it will be `DBAL <https://github.com/doctrine/dbal>`_
--  Clone your fork locally
+-  Make the local repository aware of your fork.
 
 .. code-block:: console
 
-    $ git clone git@github.com:username/dbal.git
+    $ git clone --origin doctrine git@github.com:doctrine/dbal.git
 
--  Enter the dbal directory and add the **doctrine** remote
+At that point, you have a repository, and you should be able to start
+doing changes. Later on, if you end up with something you are satisfied
+with, you will need to be able to push your changes to your fork.
+
+Here is how you can make your local repository aware of your fork:
 
 .. code-block:: console
 
     $ cd dbal
-    $ git remote add doctrine git://github.com/doctrine/dbal.git
+    $ git remote add fork git@github.com:username/dbal.git
 
-Branching from the default branch
----------------------------------
+You should now have two remotes: one called ``doctrine``, another called
+``fork``.
```

Comment:
This should be familiar to anyone who has contributed on GitHub. For first time contributors, there is probably enough material out there on how to get started, including GitHub.com documentation. 

IMHO we do not need to write/maintain this information ourselves.

Reply:
Are you suggesting I remove this first paragraph, or replace it with a link? If the latter which one? This, maybe: https://docs.github.com/en/get-started/quickstart/github-flow ?

Reply:
I'd say https://docs.github.com/en/get-started/quickstart/contributing-to-projects explains how to fork a repo, clone the fork, create a branch, push it and make a PR.

https://docs.github.com/en/get-started/quickstart/hello-world is even more basic and goes through GitHub fundamentals.

In general, it's really hard to tell what would help and invite people with little to no experience to contribute, when you've been using Git or GitHub for a long time. I tend to forget about how it was to learn.

Reply:
✅